woocommerce/woocommerce-ios
# [Tablet Orders] Sync removal from order form to product selector

Closes: #11763


## Description


When we remove an item using the button in the collapsible product row on the order form, we may need to update the product selector to keep the selection state in sync when both views are shown on screen.

In this PR, we add `syncChangesImmediately` to the view model, which is based on the same `presentationStyle == .sideBySide` check as we used in #11905. To access the `presentationStyle`, we had to make `OrderForm` a strict child view of the `AdaptiveModalContainer`, which was most easily achieved using a wrapper view.

When the changes should be synced immediately, we call `removeSelection` on the product selector view model as well.

This PR has reinforced my view that we need to significantly refactor the order logic. I think we need a shared data source for the order we're building, rather than trying to sync changes back and forth between two views and the network. That's for a future change, however.

## Testing instructions


1. Enable the `sideBySideViewForOrderForm` feature flag
2. Launch the app on an iPad, and go to the `Orders` tab
3. Tap `+`
4. Ensure that you can see the side-by-side view in your current layout
5. Add a product to the order
6. Expand the product row in the order summary on the right hand side
7. Tap `Remove product from order`
8. Observe that the product or variation is deselected in the product selector as well

Repeat with editing, and different screen splits (compact vs regular layouts.)

Check that you can dismiss the order, and take payment directly from the order form.

Turn the feature off and check dismissal and `Collect Payment` again.

## Screenshots


https://github.com/woocommerce/woocommerce-ios/assets/2472348/e69d521f-b5c4-4578-b4a9-4d862b4ca100



---
